src/elevator/ElevatorQueue.java

```diff
@@ -0,0 +1,105 @@
+package elevator;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+
+import messages.ElevatorRequestMessage.Direction;
+
+public class ElevatorQueue {
+
+	private LinkedList<Integer> pickUpQueue;
+	private LinkedList<Integer> dropOffQueue;
+	private Direction direction; 
+	
+	public ElevatorQueue() {
+		pickUpQueue = new LinkedList<Integer>();
+		dropOffQueue = new LinkedList<Integer>();
+	}
+	
+	public boolean isEmpty() {
+		return (pickUpQueue.isEmpty() && dropOffQueue.isEmpty());
+	}
+	
+	public void addPickUp(Integer i) {
+		pickUpQueue.add(i);
+	}
+	
+	public void addDropOff(Integer i) {
+		dropOffQueue.add(i);
+	}
+	
+	public int size() {
+		return pickUpQueue.size() + dropOffQueue.size();
+	}
+	
+	public int peek() {
+		if (isEmpty()) {
+			return -1;
+		}
+		if (pickUpQueue.isEmpty()) {
+			return dropOffQueue.peek();
+		} else if (dropOffQueue.isEmpty()) {
+			return pickUpQueue.peek();
+		} 
+		
+		if (direction == Direction.UP) {
+			return Math.min(pickUpQueue.peek(), dropOffQueue.peek());
+		} else {
+			return Math.max(pickUpQueue.peek(), dropOffQueue.peek());
+		}
+	}
+	
+	public int pickUpPeek() {
+		if (pickUpQueue.isEmpty()) {
+			return -1;
+		}
+		return pickUpQueue.peek();
+	}
+	
+	public int dropOffPeek() {
+		if (dropOffQueue.isEmpty()) {
+			return -1;
+		}
+		return dropOffQueue.peek();
+	}
+	
+	public Integer remove() {
+		int peek = peek();
+		if (peek != -1) {
+			while (peek == pickUpPeek()) {
+				pickUpQueue.remove();
+			}
+			while (peek == dropOffPeek()) {
```

Comment:
you don't actually need the `while` functionality here, right? An`if` statement would work instead?

Reply:
This is intended to account for multiple of the same pick up and drop off (if the queue looks like {1, 2, 2, 2, 3, 5} for example).